cloudbridge/cloud/providers/openstack/provider.py

```diff
@@ -236,19 +238,48 @@ def _connect_cinder(self):
 #         return glance_client.Client(version=api_version,
 #                                     session=self.keystone.session)
 
-    def _connect_swift(self):
+    @staticmethod
+    def _clean_options(options, method_to_match):
+        """ 
+        :param options: The source options.
+        :param method_to_match: The method whose signature is to be matched
+        :return: A copy of the source options with all keys that are not in the
+            ``method_to_match`` parameter list removed. If options is ``None`` 
+            then will be an empty dictionary
+        """
+        result = dict(options or {})
+        if len(result):
+            # Don't allow the options to override our authentication
+            result['os_options'] = None
+            passed_in_options = set(result.keys())
+            method_signature = inspect.signature(method_to_match)
+            parameters = set(method_signature.parameters.keys())
+            difference = passed_in_options - parameters
+            for name in difference:
+                del result[name]
+        return result
+
+    def connect_swift(self, options=None):
```

Comment:
Yes, this is now being called from the outside class but I'm a bit hesitant to make this a public method. It adds asymmetry between the providers and may introduce a wrong impression on how methods should be called. So I'm thinking we may be better off leaving this private regardless of the outside call. @nuwang?